### PR TITLE
New plugins property to pass mmap buffer

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Replacement for https://github.com/openvinotoolkit/openvino/pull/27644

### Tickets:
 - CVS-154602
 - CVS-157192

### CODE REVIEW

File: src/inference/src/dev/core_impl.cpp [hunk 1]
```diff
@@ -1459,6 +1459,9 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::load_model_from_cache(
                         update_config[ov::weights_path.name()] = weights_path;
                     }
                 }
+                if (model_buffer) {
+                    update_config[ov::internal::cached_model_buffer.name()] = model_buffer;
```
Comment (ilya-lavrenov) [inline src/inference/src/dev/core_impl.cpp:1463]: should we populate `update_config` only iff plugin support import with mmap?

File: src/plugins/intel_cpu/src/plugin.cpp [hunk 2]
```diff
@@ -565,8 +565,13 @@ std::shared_ptr<ov::ICompiledModel> Plugin::import_model(std::istream& model_str
         decript_from_string = true;
     }
 
+    std::shared_ptr<ov::AlignedBuffer> model_buffer;
```
Comment (ilya-lavrenov) [inline src/plugins/intel_cpu/src/plugin.cpp:568]:
```suggestion
    std::shared_ptr<ov::AlignedBuffer> model_buffer = nullptr;
```

File: src/inference/src/dev/core_impl.cpp (hunk 1 above)
Comment (olpipi) [reply src/inference/src/dev/core_impl.cpp:1463]: I think no. Plugins don't expect they will get one more extra element in config. So we have to update all plugins in such case.

File: src/plugins/intel_cpu/src/plugin.cpp (hunk 2 above)
Comment (olpipi) [reply src/plugins/intel_cpu/src/plugin.cpp:569]: shared_ptr default ctr assigns it to nullptr

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**